### ISSUE TITLE
Watchdog stopping Outlook express from speaking

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2523]
**Reported by briang1 on 2012-07-12 10:43**
It appears that on the first run of Outlook express in a session, Watchdog activity is triggered the first time t you hit enter to read a message in plain text. Thus although Outlook express is displaying the email nvda remains silent even though it was reading the Inbox befor. The cure is to escape and reselect the message but its a bit annoying so i thought it best to report the problem.

Nothing will work on the opened message in the above state, not even say all. Here in log extract 1, we see the effect when nvda tries to open the message.

IO - inputCore.InputManager.executeGesture (09:46:41):
Input: kb(desktop):upArrow
IO - speech.speak (09:46:41):
Speaking [Amess  Subject: Re: Dead Right, Received: 11/07/2012 11:38  1112 of 1113'](u'Barbara)
IO - inputCore.InputManager.executeGesture (09:46:43):
Input: kb(desktop):downArrow
IO - speech.speak (09:46:43):
Speaking [Amess  Subject: Re: Dead Right, Received: 11/07/2012 15:13  1113 of 1113'](u'Barbara)
IO - inputCore.InputManager.executeGesture (09:46:43):
Input: kb(desktop):enter
DEBUGWARNING - watchdog._watcher (09:46:44):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 155, in <module>
  File "core.pyc", line 339, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 316, in Notify
  File "IAccessibleHandler.pyc", line 832, in pumpAll
  File "IAccessibleHandler.pyc", line 724, in processForegroundWinEvent
  File "IAccessibleHandler.pyc", line 509, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyc", line 38, in getNVDAObjectFromEvent
  File "IAccessibleHandler.pyc", line 336, in accessibleObjectFromEvent
  File "oleacc.pyc", line 240, in AccessibleObjectFromEvent

DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (09:46:45):
oleacc.AccessibleObjectFromEvent with window 328646, objectID -4 and childID 0: 
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (09:46:46):
oleacc.AccessibleObjectFromEvent with window 328646, objectID -4 and childID 0: 
DEBUGWARNING - watchdog._watcher (09:46:46):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 155, in <module>
  File "core.pyc", line 339, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 316, in Notify
  File "IAccessibleHandler.pyc", line 815, in pumpAll
  File "IAccessibleHandler.pyc", line 629, in processFocusWinEvent
  File "IAccessibleHandler.pyc", line 509, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyc", line 38, in getNVDAObjectFromEvent
  File "IAccessibleHandler.pyc", line 336, in accessibleObjectFromEvent
  File "oleacc.pyc", line 240, in AccessibleObjectFromEvent

DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (09:46:47):
oleacc.AccessibleObjectFromEvent with window 328646, objectID -4 and childID 0: 
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (09:46:48):
oleacc.AccessibleObjectFromEvent with window 66520, objectID -4 and childID 1: 
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (09:46:49):
oleacc.AccessibleObjectFromEvent with window 328646, objectID -4 and childID 0: 
DEBUGWARNING - watchdog._watcher (09:46:52):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 155, in <module>
  File "core.pyc", line 339, in main
  File "wx_core.pyc", line 8010, in MainLoop
  File "wx_core.pyc", line 7306, in MainLoop
  File "core.pyc", line 316, in Notify
  File "IAccessibleHandler.pyc", line 829, in pumpAll
  File "IAccessibleHandler.pyc", line 587, in processGenericWinEvent
  File "IAccessibleHandler.pyc", line 509, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible__init__.pyc", line 38, in getNVDAObjectFromEvent
  File "IAccessibleHandler.pyc", line 336, in accessibleObjectFromEvent
  File "oleacc.pyc", line 240, in AccessibleObjectFromEvent

IO - inputCore.InputManager.executeGesture (09:46:54):
Input: kb(desktop):NVDA+downArrow
DEBUG - queueHandler.registerGeneratorObject (09:46:54):
Adding generator 3
DEBUG - queueHandler.pumpAll (09:46:54):
generator 3 finished
IO - inputCore.InputManager.executeGesture (09:47:10):
Input: kb(desktop):control+alt+i
INFO - core.main (09:47:11):
Exiting

Disabling watchdog and  doing the same, ie resetting the computer so its a first run again, the operation works as expected and you can read the message. the delay is much the same, but it does work. Here is the log from that..

IO - inputCore.InputManager.executeGesture (11:07:35):
Input: kb(desktop):upArrow
IO - speech.speak (11:07:35):
Speaking [Amess  Subject: Re: Dead Right, Received: 11/07/2012 15:13  1113 of 1114'](u'Barbara)
IO - inputCore.InputManager.executeGesture (11:07:39):
Input: kb(desktop):enter
IO - speech.speak (11:07:41):
Speaking [Dead Right - Unicode (UTF-8)'](u'Re:)
IO - speech.speak (11:07:41):
Speaking [ multi line  read only'](u'edit)
IO - speech.speak (11:07:41):
Speaking [Linda not Helen.  She is just like an old school '](u'Sorry,)
IO - inputCore.InputManager.executeGesture (11:07:44):
Input: kb(desktop):NVDA+downArrow
DEBUG - queueHandler.registerGeneratorObject (11:07:44):
Adding generator 35
IO - speech.speak (11:07:44):
Speaking [u'Sorry, Linda not Helen.  '](IndexCommand%281%29,)
IO - speech.speak (11:07:44):
Speaking [is just like an old school ', IndexCommand(2), u'friend of mine \r', IndexCommand(3), u'called Helen and I always have to think carefully ', IndexCommand(4), u'when I see her.  '](u'She)
IO - speech.speak (11:07:46):
Speaking [IndexCommand(5), u'Barb\r', IndexCommand(6), u'----- Original Message ----- \r', IndexCommand(7), u'From: " 1brian Gaff" briang1@blueyonder.co.uk\r', IndexCommand(8), u'To: "Barbara Amess" barxxx@xxx.com\r', IndexCommand(9), u'Sent: Wednesday, July 11, 2012 3:01 PM\r', IndexCommand(10), u'Subject: Re: Dead Right\r', IndexCommand(11), IndexCommand(12), IndexCommand(13), u"> Helen, I don't remember a helen?\r"](u'Whoops!!!r',)
IO - inputCore.InputManager.executeGesture (11:07:49):
Input: kb(desktop):escape
IO - speech.speak (11:07:49):
Speaking [\- Outlook Express - Brian's main account"](u"Inbox)
IO - speech.speak (11:07:49):
Speaking [Express Message List  list'](u'Outlook)
IO - speech.speak (11:07:49):
Speaking [Amess  Subject: Re: Dead Right, Received: 11/07/2012 15:13  1113 of 1114'](u'Barbara)
IO - inputCore.InputManager.executeGesture (11:08:03):
Input: kb(desktop):control+alt+i
INFO - core.main (11:08:03):
Exiting

I'm not sure quite why its  saying its  frozen, and indeed why the system is not seeing the message open when watchdog is active.

**Comment by leonardder** [comment]
Outlook Express is Abandoned software. Closing as won't fix.

**Comment by Brian1Gaff** [comment]
Thats as maybe but i still use it. I've had no issues with the latest version of Outlook Express on the current nvda versions even in windows 10 with the mshtml special file system now out there for it.
 I'd advise not  removing app modules though as many like myself like the program for special reasons.
 Brian

bglists@blueyonder.co.uk 
Sent via blueyonder. 
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
This message sent from a Windows XP machine!
  ----- Original Message -----

**Comment by leonardder** [comment]
I think it's not being planned to remove appmodules for legacy apps at all, at least not until XP support is officially dropped. Anyway, I'll do a works for me instead.
